### PR TITLE
RHCLOUD-30714 Update References from "Inventory Groups" to "Workspaces"

### PR DESCRIPTION
**Pull request by aferd**
### Description

Update References from "Inventory Groups" to "Workspaces"

[RHCLOUD-30714](https://issues.redhat.com/browse/RHCLOUD-30714)

---

#### Before:
Screenshot 2024-04-25 at 12 42 31 PM

#### After:
Screenshot 2024-04-25 at 12 35 51 PM

---

### Checklist ☑️
- [X] PR only fixes one issue or story 
- [X] Change reviewed for extraneous code 
- [X] UI best practices adhered to 
- [X] Commits squashed and meaningfully named 
- [ ] All PR checks pass locally (build, lint, test, E2E)

##
- [ ] _(Optional) QE: Needs QE attention (OUIA changed, perceived impact to tests, no test coverage)_
- [ ] _(Optional) QE: Has been mentioned_
- [ ] _(Optional) UX: Needs UX attention (end user UX modified, missing designs)_
- [ ] _(Optional) UX: Has been mentioned_
##

### CODE REVIEW

File: src/Messages.js [hunk 1]
```diff
@@ -701,8 +701,8 @@ export default defineMessages({
   },
   groupDefinition: {
     id: 'groupDefinition',
-    description: 'Group definition label',
```
Comment (fhlavac) [inline src/Messages.js:704]: We could perhaps update also message identifiers to separate them from the regular RBAC groups

**Review by fhlavac — DISMISSED** [review]
@aferd looks great! Just one thought about changing the message identifiers for clarity. What do you think?

File: src/Messages.js [hunk 2]
```diff
@@ -700,7 +700,7 @@ export default defineMessages({
     defaultMessage: 'Group description',
   },
   groupDefinition: {
```
Comment (fhlavac) [inline src/Messages.js:702]: @aferd sorry, one last request, could you rename also these keys to match the IDs? Thank you :-)

**Comment by karelhala** [comment]
Is this hidden behind feature flag? So we don't scare customers when we release to prod.

File: src/Messages.js (hunk 2 above)
Comment (aferd) [reply src/Messages.js:702]: oh, forgot to push that up! doing it now.

File: src/helpers/role/inventory-helper.js [hunk 3]
```diff
@@ -8,6 +9,8 @@ export const getInventoryGroups = async ({ name, perPage, page, options } = {})
   return await inventoryResourceTypesApi.apiResourceTypeGetResourceTypeGroupsList(name, perPage, page, options);
 };
 
+export const enableWorkspacesNameChange = useFlag('platform.rbac.groups-to-workspaces-rename');
```
Comment (fhlavac) [inline src/helpers/role/inventory-helper.js:12]: Hooks should be called only in function components https://react.dev/reference/rules/rules-of-hooks

File: src/smart-components/role/add-role/schema.js [hunk 4]
```diff
@@ -13,11 +13,13 @@ import messages from '../../../Messages';
 import providerMessages from '../../../locales/data.json';
 import { validateNextAddRolePermissionStep } from '../permission-wizard-helper';
 import InventoryGroupsRoleTemplate from './inventory-groups-role-template';
-import { enableWorkspacesNameChange } from '../../../helpers/role/inventory-helper';
+import { useFlag } from '@unleash/proxy-client-react';
 
 export const schemaBuilder = (container) => {
   const cache = createIntlCache();
   const intl = createIntl({ locale, messages: providerMessages }, cache);
+  const enableWorkspacesNameChange = useFlag('platform.rbac.groups-to-workspaces-rename');
```
Comment (fhlavac) [inline src/smart-components/role/add-role/schema.js:21]: The hook should not be used here. Please, let's pass the flag value as a prop instead.

**Review by fhlavac — APPROVED** [review]

**Pull request merged**